tests/yast2_gui/bootloader/bootcode_options.pm

```diff
@@ -0,0 +1,70 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+#
+# Summary: Open bootloader gui, verify default setting in ui, change
+# some settings and verify that they have been applied.
+#
+# Maintainer: QE YaST <qa-sle-yast@suse.de>
+
+use base "y2_module_guitest";
+use strict;
+use warnings;
+use testapi;
+use scheduler 'get_test_suite_data';
+use YaST::Module;
+use cfg_files_utils;
+use x11utils 'start_root_shell_in_xterm';
+use y2lan_restart_common 'close_xterm';
+
+my $bootloader;
+
+sub verify_current_options {
+    my $test_data        = get_test_suite_data();
+    my %current_settings = $bootloader->get_current_settings();
+    compare_settings({expected => $test_data->{bootloader}->{bootcode_options}, current => \%current_settings});
+}
+
+sub save_screenshot_and_close {
+    my $firstboot = $testapi::distri->get_firstboot();
+    wait_still_screen;
+    save_screenshot;
+    $firstboot->press_next();
+    YaST::Module::close(module => 'bootloader');
+}
+
+sub close_check_open {
+    my ($args) = @_;
+    save_screenshot_and_close();
+    grep_file_and_report({
+            must_match     => $args->{must_match},
+            must_not_match => $args->{must_not_match},
+            config_file    => "/etc/default/grub_installdevice"});
+    assert_script_run("dd if=/dev/vda bs=512 count=1 | hexdump -C | grep GRUB $args->{grep_arg}");
+    YaST::Module::open(module => 'bootloader', ui => 'qt') unless $args->{noopen};
+}
+
+sub verify_applied_config {
+    my ($self) = @_;
+    close_check_open({must_match => ['by-path', 'activate']});
+    $bootloader->toggle_checkboxes();
+    close_check_open({must_match => ['by-uuid', 'by-path', 'generic_mbr'], must_not_match => ['activate']});
+    $bootloader->uncheck_write_to_mbr();
```

Comment:
instead of 'uncheck_write_to_mbr' to not talk about UI controls, better 'not_write_mbr', for example.

Reply:
I really don't understand this idea to no talk about controls

Reply:
not sure if I get your answer, but next() could be a method in the controller, it could be called accept_changes() or something like that, even next() could be valid. Still I can only see OK and Cancel button in the screen. We might have a quick sync offline to clarify remaining things.